src/multilayer_canopy/MLCanopyFluxesMod.F90

```diff
@@ -0,0 +1,1114 @@
+    swleaf         => mlcanopy_inst%swleaf_leaf              , &  ! Leaf absorbed solar radiation (W/m2 leaf)
+    lwleaf         => mlcanopy_inst%lwleaf_leaf              , &  ! Leaf absorbed longwave radiation (W/m2 leaf)
+    rnleaf         => mlcanopy_inst%rnleaf_leaf              , &  ! Leaf net radiation (W/m2 leaf)
+    tleaf          => mlcanopy_inst%tleaf_leaf               , &  ! Leaf temperature (K)
+    tleaf_bef      => mlcanopy_inst%tleaf_bef_leaf           , &  ! Leaf temperature for previous timestep (K)
+    tleaf_hist     => mlcanopy_inst%tleaf_hist_leaf          , &  ! Leaf temperature (not sun/shade average) for history files (K)
+    lwp            => mlcanopy_inst%lwp_leaf                 , &  ! Leaf water potential (MPa)
+    lwp_hist       => mlcanopy_inst%lwp_hist_leaf              &  ! Leaf water potential (not sun/shade average) for history files (MPa)
+    )
+
+    ! Get current step counter (nstep) and step size (dtime)
+
+    nstep = get_nstep()
+    dtime = get_step_size()
+
+    ! Set number of sub-time steps for flux calculations
+
+    num_sub_steps = int(dtime / dtime_substep)
+
+    ! Build filter of patches to process with multilayer canopy.
+    ! For now only process tall canopies > 0.5 m
+
+    num_mlcan = 0
+    do fp = 1, num_exposedvegp
+       p = filter_exposedvegp(fp)
+       g = patch%gridcell(fp)
+!      if (grc%latdeg(g) .gt. -2.9_r8 .and. grc%latdeg(g) .lt. -2.7_r8) then
+!      if (grc%londeg(g) .gt. 294.5_r8 .and. grc%londeg(g) .lt. 295.5_r8) then
+       if (htop(p) .ge. 0.5) then
+          num_mlcan = num_mlcan + 1
+          filter_mlcan(num_mlcan) = p
+       end if
+!      end if
+!      end if
+    end do
+
+    ! Initialize canopy vertical structure and profiles. This is only done
+    ! once (on first time step) because the forcing height (and therefore
+    ! the volume of air in the surface layer) changes between time steps.
+    ! The call for initialization is triggered by zref = spval.
+
+    ml_vert_init = 0
+    do fp = 1, num_mlcan
+       p = filter_mlcan(fp)
+       if (zref(p) == spval) ml_vert_init = 1
+    end do
+
+    if (ml_vert_init == 1) then
+       if (masterproc) then
+          write (iulog,*) 'Attempting to initialize multilayer canopy vertical structure .....'
+       end if
+
+       call initVerticalStructure (bounds, num_mlcan, filter_mlcan, &
+       canopystate_inst, frictionvel_inst, mlcanopy_inst)
+
+       call initVerticalProfiles (num_mlcan, filter_mlcan, &
+       atm2lnd_inst, wateratm2lndbulk_inst, mlcanopy_inst)
+
+       if (masterproc) then
+          write (iulog,*) 'Successfully initialized multilayer canopy vertical structure'
+       end if
+    end if
+
+    ! Copy CLM variables to multilayer canopy variables. Note the
+    ! distinction between grid cell (g), column (c), and patch (p)
+    ! variables. All multilayer canopy variables are for patches.
+
+    do fp = 1, num_mlcan
```

Comment:
Debugging:
Focus on this loop and next ending with line 417.

Reply:
This far the "write statements" seem to point to tleaf(p=1, ic=9), which increases out of control from 285 to 1264K by nstep = 10. The trend in tleaf(1,9)  is already appearent from nstep > 3.

Likely related... I see lw_source < 0 for ic = 9 and 10. From the tiny fracsun I infer that ic = 9 is nearest to the ground.

What I don't know how to interpret is whether the tridiagonal matrix solver has anything to do with the problem.

Gordon, it may help if you look at the output that I'm looking at. 
- Open the last lnd.log file in:
/glade/derecho/scratch/slevis/US-NR1_sp/run
- The write statements are in
``` 
/glade/work/slevis/git/multilayer_canopy/src/multilayer_canopy/src/multilayer_canopy/MLCanopyFluxesMod.F90
/glade/work/slevis/git/multilayer_canopy/src/multilayer_canopy/src/multilayer_canopy/MLLongwaveRadiationMod.F90
```
I think you can type `git diff` in /glade/work/slevis/git/multilayer_canopy to see clearly what I did.